### PR TITLE
Initial API app integration and issue #15 

### PR DESCRIPTION
**Pull request by Gerdusvdl**
# Description of changes

## What?
This PR is mainly to include the models that will be used for the survey app, as well as the initial API application. There is integration between the api_app and the survey_design app, but at the moment **only for reading data from the database** . The api_app includes the following new file:
- `serializers.py`  - Created per-model for exposing data fields to the API

Basically the API is linked to the database backend, and the routing URL paths that can be used to access them are defined, along with the fields that are exposed to the API. Internally, however, the models are imported and accessed directly to save computational time. See comments in `apiapp\views.py`

## Why?
The PR is done to ensure all developers can work simultaneously and not risk too many eventual merge conflicts. In favour of speed, this PR will be done first with only API reading capabilities and the writing and updating only done later, so that other developers do not have to wait on this PR to continue.

## Note:
References have been added. Unit testing code will also still be included in later updates.

### CODE REVIEW

File: citizenvoice/apiapp/models/answer.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+# Represents a single answer given to a certain question as part of a user's response
+class Answer(models.Model):
+    response = models.ForeignKey(Response, on_delete=models.CASCADE)
+    question = models.ForeignKey(Question, on_delete=models.CASCADE)
+    created = models.DateTimeField(_("Creation date"))
+    updated = models.DateTimeField(_("Last edited"))
+    body = models.TextField(_("Answer Body"))
+    lon = models.FloatField()
+    lat = models.FloatField()
```
Comment (manuGil) [inline citizenvoice/apiapp/models/answer.py:22]: Add an empty line to the end of the file. Do the same with the rest of the source code files

File: citizenvoice/apiapp/models/question.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+This code is based on the source code of the django-survey application
+by Pierre Sassoulas, 2022, version 1.4.0. 
+Available at https://github.com/Pierre-Sassoulas/django-survey
+"""
+
+from django.db import models
+from django.utils.translation import gettext_lazy as _
+from .survey import Survey
+
+# Represents a single question of any type
```
Comment (manuGil) [inline citizenvoice/apiapp/models/question.py:11]: Add class purpose as DockStrings. Do the same for all classes.

File: citizenvoice/citizenvoice/settings.py [hunk 3]
```diff
@@ -95,14 +97,21 @@
 # Database
 # https://docs.djangoproject.com/en/4.0/ref/settings/#databases
 
+# DATABASES = {
```
Comment (manuGil) [inline citizenvoice/citizenvoice/settings.py:100]: Keep the configuration for the Postgres database. Sqlite3 doesn't support spatial types, and the spatialite extension gives me many errors when migrating.

File: citizenvoice/survey_design/models/__init__.py [hunk 4]
```diff
@@ -1,5 +0,0 @@
-from .answer import Answer
```
Comment (manuGil) [inline citizenvoice/survey_design/models/__init__.py:1]: Did you delete this file? Keep it even if it is empty

File: citizenvoice/apiapp/templates/survey/base.html [hunk 5]
```diff
@@ -0,0 +1,17 @@
+{% load static %}
```
Comment (manuGil) [inline citizenvoice/apiapp/templates/survey/base.html:1]: Are the HTML templates necessary? The API doesn't need to render data in HTML. What is the reason behind this?

**Review by manuGil — CHANGES_REQUESTED** [review]
Please, start making the changes in the comments. I will review the rest tomorrow.

File: citizenvoice/survey_design/models/__init__.py (hunk 4 above)
Comment (Gerdusvdl) [reply citizenvoice/survey_design/models/__init__.py:1]: Yes, these models are not/ should not be used anymore, as the API integration replaces this. Should I rather not delete unused files/ files that should no longer be used in the future?

File: citizenvoice/citizenvoice/settings.py (hunk 3 above)
Comment (Gerdusvdl) [reply citizenvoice/citizenvoice/settings.py:100]: As far as I know, we finally decided on our meeting on the 15th of August to only use SQLite for now. So should I still include it but leave it commented out?

File: citizenvoice/apiapp/templates/survey/base.html (hunk 5 above)
Comment (Gerdusvdl) [reply citizenvoice/apiapp/templates/survey/base.html:1]: This was only for debugging for myself, I completely forgot to delete them!

File: citizenvoice/survey_design/models/__init__.py (hunk 4 above)
Comment (manuGil) [reply citizenvoice/survey_design/models/__init__.py:1]: Please, keep it to maintain the Django structure of an app

File: citizenvoice/apiapp/urls.py [hunk 6]
```diff
@@ -0,0 +1,25 @@
+# ====================================================================================================================
+#
```
Comment (manuGil) [inline citizenvoice/apiapp/urls.py:2]: I checked that the api root works, but I not the paths to questions and such. Maybe you can add some examples of how to use the API in the next pull request. 

Screenshot from 2022-08-29 11-42-16
Screenshot from 2022-08-29 11-43-15

**Review by manuGil — APPROVED** [review]

**Pull request merged**